src/battle.ts

```diff
@@ -1662,6 +1663,7 @@ export class Battle {
 					this.scene.runStatusAnim('cursed' as ID, [poke]);
 					break;
 				case 'confusion':
+					poke.timesAttacked += 1;
```

Comment:
confusion hits don't count for rage fist, so you can get rid of this

Reply:
Smogon says it does: https://www.smogon.com/dex/sv/moves/rage-fist/. Is Smogon incorrect?